File: apps/customers-service/app/schemas/customer.py
"""Customer Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel, EmailStr, Field


class CustomerBase(BaseModel):
    """Base customer schema with common fields."""

    name: str = Field(..., min_length=1, max_length=255, description="Customer name")
    email: EmailStr = Field(..., description="Customer email address")
    phone: str | None = Field(None, max_length=50, description="Customer phone number")


class CustomerCreate(CustomerBase):
    """Schema for creating a new customer."""

    pass


class CustomerResponse(CustomerBase):
    """Schema for customer response."""

    id: int = Field(..., description="Customer ID")
    created_at: datetime = Field(..., description="Customer creation timestamp")

    model_config = {"from_attributes": True}


# Alias for backward compatibility
Customer = CustomerResponse

File: apps/mfe-customers/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  email: string;
  phone: string | null;
  created_at: string;
}

export interface CustomerCreate {
  name: string;
  email: string;
  phone?: string;
}

export interface LoanRequest {
  customer_id: number;
  amount: string;
  interest_rate: string;
  term_months: number;
}


